core/sr-primitives/src/lib.rs

```diff
@@ -122,6 +122,61 @@ impl From<f32> for Permill {
 	}
 }
 
+/// Perbill is parts-per-billion. It stores a value between 0 and 1 in fixed point and
+/// provides a means to multiply some other value by that.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+#[derive(Encode, Decode, Default, Copy, Clone, PartialEq, Eq)]
+pub struct Perbill(u32);
+
+// TODO: impl Mul<Perbill> for N where N: As<usize>
+impl Perbill {
+	/// Attenuate `b` by self.
+	pub fn times<N: traits::As<usize> + ::rstd::ops::Mul<N, Output=N> + ::rstd::ops::Div<N, Output=N>>(self, b: N) -> N {
+		// TODO: handle overflows
+		b * <N as traits::As<usize>>::sa(self.0 as usize) / <N as traits::As<usize>>::sa(1_000_000_000)
```

Comment:
this code will be used in WASM and native, so using usize here could cause inconsistencies